src/debug_utils.cc

```diff
@@ -145,36 +215,35 @@ class Win32SymbolDebuggingContext final : public NativeSymbolDebuggingContext {
     return CaptureStackBackTrace(0, count, frames, nullptr);
   }
 
+  Win32SymbolDebuggingContext() = delete;
+  Win32SymbolDebuggingContext(const Win32SymbolDebuggingContext&) = delete;
+  Win32SymbolDebuggingContext(Win32SymbolDebuggingContext&&) = delete;
+  Win32SymbolDebuggingContext operator=(const Win32SymbolDebuggingContext&)
+    = delete;
+  Win32SymbolDebuggingContext operator=(Win32SymbolDebuggingContext&&)
+    = delete;
+
  private:
   HANDLE current_process_;
+  FILE* out_;
 };
 
 std::unique_ptr<NativeSymbolDebuggingContext>
 NativeSymbolDebuggingContext::New() {
   return std::unique_ptr<NativeSymbolDebuggingContext>(
-      new Win32SymbolDebuggingContext());
+      new Win32SymbolDebuggingContext(stderr));
 }
 
 #endif  // __POSIX__
 
-std::string NativeSymbolDebuggingContext::SymbolInfo::Display() const {
-  std::string ret = name;
-  if (!filename.empty()) {
-    ret += " [";
-    ret += filename;
-    ret += ']';
-  }
-  return ret;
-}
-
 void DumpBacktrace(FILE* fp) {
   auto sym_ctx = NativeSymbolDebuggingContext::New();
   void* frames[256];
   const int size = sym_ctx->GetStackTrace(frames, arraysize(frames));
   for (int i = 1; i < size; i += 1) {
     void* frame = frames[i];
-    fprintf(fp, "%2d: %p %s\n",
-            i, frame, sym_ctx->LookupSymbol(frame).Display().c_str());
+    auto s = sym_ctx->LookupSymbol(frame);
+    fprintf(fp, "%2d: %p %s\n", i, frame, s.Display().c_str());
```

Comment:
Unrelated change? Imo this makes thing a bit more confusing, but if you really want to split it, make the type of `s` explicit.

Reply:
This is an interesting case. I didn't remove existing information, so There's no reduction of readability...
I'd rather keep ignored the type of s, having it this way allows the compiler to elide it with optimization, while still have it in debug mode.

Reply:
> There's no reduction of readability...

That’s subjective, but it’s not the end of the world here.

> I'd rather keep ignored the type of s, having it this way allows the compiler to elide it with optimization, while still have it in debug mode.

I kinda do not understand this sentence. Why would this affect compiler optimizations, or what difference does debug mode make?

Reply:
using `auto` is a hint to the compiler that we are not using any property of the variable other then those explicitly stated.
But I'll fix it... It's not a hot path, or a big deal...

Reply:
`auto` just means type inference, and code generation happens after types have been inferred, so there should not usually be *any* difference in optimization if we explicitly spell out types.